user/views.py

```diff
@@ -1 +1,25 @@
-# write your code here
+from rest_framework import generics
+from rest_framework.authentication import TokenAuthentication
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.settings import api_settings
+from rest_framework.authtoken.views import ObtainAuthToken
+
+from user.permissions import IsAdminOrIfAuthenticatedReadOnly
+from user.serializers import UserSerializer
+
+
+class CreateUserView(generics.CreateAPIView):
+    serializer_class = UserSerializer
+
+
+class LoginUserView(ObtainAuthToken):
+    renderer_classes = api_settings.DEFAULT_RENDERER_CLASSES
+
+
+class ManageUserView(generics.RetrieveUpdateAPIView):
+    serializer_class = UserSerializer
+    authentication_classes = (TokenAuthentication,)
+    permission_classes = (IsAuthenticated,)
```

Comment:
The `ManageUserView` should use the `IsAdminOrIfAuthenticatedReadOnly` permission class instead of `IsAuthenticated` to meet the task requirements for access control.